daisy/workflow/workflow.go

```diff
@@ -223,20 +238,52 @@ func (w *Workflow) populateStep(step *Step) error {
 	step.SubWorkflow.workflow.StorageClient = w.StorageClient
 	step.SubWorkflow.workflow.Ctx = w.Ctx
 	step.SubWorkflow.workflow.Cancel = w.Cancel
-	step.SubWorkflow.workflow.logger = w.logger
+	step.SubWorkflow.workflow.gcsLogWriter = w.gcsLogWriter
+	step.SubWorkflow.workflow.vars = map[string]vars{}
 	for k, v := range step.SubWorkflow.Vars {
-		step.SubWorkflow.workflow.Vars[k] = v
+		step.SubWorkflow.workflow.vars[k] = vars{Value: v}
 	}
 	return step.SubWorkflow.workflow.populate()
 }
 
+func (w *Workflow) populateVars() error {
+	if w.vars == nil {
+		w.vars = map[string]vars{}
+	}
+	for k, v := range w.Vars {
+		// Don't overwrite existing vars (applies to subworkflows).
+		if _, ok := w.vars[k]; ok {
+			continue
+		}
+		var sv string
+		if err := json.Unmarshal(v, &sv); err == nil {
+			w.vars[k] = vars{Value: sv}
+			continue
+		}
+		var vv vars
```

Comment:
This isn't plural, well 'vars' is but that may be a bad struct name. sv above was string value, this was vars value